velox/vector/fuzzer/VectorFuzzer.h

```diff
@@ -232,7 +232,8 @@ class VectorFuzzer {
     return boost::random::uniform_01<double>()(rng_) < n;
   }
 
-  // Wraps the given vector in a LazyVector.
+  // Wraps the given vector in a LazyVector. If there are multiple dictionary
```

Comment:
The implementation seems to handle only 2 layers of dictionary by putting LV between these. Why only 2? Why do we need to LV between layers and not directly on top?

I'm also thinking that, perhaps, we could modify the Fuzzer to run expressions over TableScan which naturally generates LVs. This would make these closer to production use case and allow for super easy repro.

Reply:
> The implementation seems to handle only 2 layers of dictionary by putting LV between these. Why only 2?

This implementation is actually recursive and will add the lazy layer on top of the innermost dict. However, after our discussion this morning I realized I forgot to factor in the fact that only a single dict layer over lazy is supported. I'll modify this to ensure a lazy layer is wrapped after the first layer in case of a multi layered dictionary.

> Why do we need to LV between layers and not directly on top?

My initial implementation combines all the dictionary layers into one similar to what a decoded vector does. This was intentional because (correct me if I am wrong) lazy vectors are generated via a scan and therefore the vectors generated from lazy mimic the data format used to store it in the file. This means it encounters at most one dictionary layer (again correct me if I am wrong, I am assuming files don’t store multiple layered dictionaries as it would be inefficient), and hence returning loaded vectors with multiple layers will not help get more coverage. If however my assumption is wrong, I'll go ahead and add support for returning multi-layered dicts from lazy vectors.

> I'm also thinking that, perhaps, we could modify the Fuzzer to run expressions over TableScan which naturally generates LVs. This would make these closer to production use case and allow for super easy repro.

Currently we generate random data in the fuzzer to feed into the expression eval, are you suggesting we generate the random data -> store it into a file, then read it via a tableScan->FilterProject pipeline?
If yes, we would have to make sure we cover all possible file storage types so that we can achieve the same variety of vectors from scans like we do currently with VectorFuzzer. This might not be possible if some combination of vector types, multiple layers of dict etc are only possible if vectors have gone through some operators.
However, I definitely agree with your last line and would love to know what kind of implementation you had in mind.